internal/network/manifests_generator.go

```diff
@@ -84,11 +84,15 @@ address=/api-int.{{.CLUSTER_NAME}}.{{.DNS_DOMAIN}}/{{.HOST_IP}}
 
 const forceDnsDispatcherScript = `
 export IP="{{.HOST_IP}}"
+export TMP_FILE="/tmp/forcedns_resolv.conf"
+
 if [ "$2" = "dhcp4-change" ] || [ "$2" = "dhcp6-change" ] || [ "$2" = "up" ] || [ "$2" = "connectivity-change" ]; then
     if ! grep -q "$IP" /etc/resolv.conf; then
-      sed -i "s/{{.CLUSTER_NAME}}.{{.DNS_DOMAIN}}//" /etc/resolv.conf
-      sed -i "s/search /search {{.CLUSTER_NAME}}.{{.DNS_DOMAIN}} /" /etc/resolv.conf
-      sed -i "0,/nameserver/s/nameserver/nameserver $IP\nnameserver/" /etc/resolv.conf
+      cp  /run/NetworkManager/resolv.conf $TMP_FILE
```

Comment:
@celebdor wdyt?

Reply:
it was @celebdor idea :)

Reply:
I don't want to take credit xD. The idea is to use NetworkManager's managed resolv.conf. However, the current implementation has an atomicity issue.

1. When creating a tmp file one should really use mktemp to prevent, among other things, different runs writing to the same place.
2. _**mv**_ is only implemented atomically with rename, which does not work across mountpoints/filesystems. Thus, the temporary file should be placed in the same directory where you want the atomic replacement to happen.

Reply:
+1

Reply:
Do we know that during the time that we copy the resolv.conf, no one is writing to it?

Reply:
That's why we copy it from /run and we hope that it should be ready there and nothing should touch it